TopologyBuilder methods return the builder to allow methods to be chained together.

rhauch opened the pull request:
Since the methods returned `void` before, this change has no impact on existing code.

ymatsuda commented:
@rhauch Thank you for the pull request. There seems to be a conflict. Would you resolve it so that I can merge?

rhauch commented:
@ymatsuda, I've rebased upon the latest upstream branch, and also added JavaDoc for the public methods in `TopologyBuilder`. Let me know if there are more problems.

ymatsuda commented:
Thanks!
